app/repositories/versioned_manual_repository.rb

```diff
@@ -0,0 +1,111 @@
+require "manual_repository"
+
+class VersionedManualRepository
+
+  class NotFoundError < StandardError; include ManualRepository::NotFoundError; end
+
+  def self.get_manual(manual_id)
+    new.get_manual(manual_id)
+  end
+
+  def get_manual(manual_id)
+    manual_record = ManualRecord.find_by(manual_id: manual_id)
+    raise NotFoundError if manual_record.nil?
+
+    {
+      draft: get_current_draft_version_of_manual(manual_record),
+      published: get_current_published_version_of_manual(manual_record),
+    }
+  end
+
+  def get_current_draft_version_of_manual(manual_record)
+    return nil if manual_record.nil?
```

Comment:
If this method isn't intended to be exposed anywhere else, I think we can remove this line as we already do checking for an nil manual in `get_manual`